attest/README.md

```diff
@@ -1,3 +1,108 @@
-## attest
+Remote attestation support. This family of crates provides support for Intel's EPID remote attestation in MobileCoin enclaves.
 
-Remote attestation primitives.
+# Crates
+
+| Name | Purpose |
+| ---- | ------- |
+| `ake` | Authenticated key exchange using the noise framework |
+| `api` | gRPC Attestation APIs and protobufs |
+| `core` | EPID remote attestation types and agnostic primitives |
+| `enclave-api` | Serializeable structures for common enclave functions |
+| `net` | IAS API Client |
+| `trusted` | Traits used to monkey-patch enclave-only functionality onto `core` types |
+| `untrusted` | Traits used to monkey-patch untrusted-only functionality onto `core` types |
+
+# Enclave Concepts and Jargon
+
+Intel's enclaves and EPID-based remote attestation is a very complex subject (there are many layers and protocols and ways to stitch things together), so the first thing to get out of the way is understanding the basics and the jargon. We'll use the MobileCoin validator (the "consensus service") for this example.
+
+## Trusted Computing Base
+
+***Trusted Computing Base*** (TCB) is pure jargon. It describes the much more pragmatic "all the things you need to believe haven't been pwned (things you trust) in order to believe the system hasn't been pwned (can be trusted)." It naturally follows that one great way to improve security is to reduce the number of things that you have to believe in to be however secure you need to be.
+
+To show how a TCB comes into being, lets take your typical "take a JSON document from HTTP, process it, return more JSON" web application, for example, isn't going to be passing raw network card buffers in RAM around from function to function. Instead you're going to have the OS deserialize the ethernet frame (and IP packet, and TCP segment), a TLS library decrypt and verify the encrypted data, a web server deserialize the HTTP request, and a framework/parser deserialize the JSON into an object, all before your application ever sees anything.
+
+And ultimately, you're going to create a TCB simply by following good development practices. Stabilizing a procedure and including it in a library that is shared in multiple places makes that library part of your TCB. You've established an ad-hoc trust in that procedure because you have used it and you "know" it works, you have unit tests around it to ensure you exercise edge cases, etc.
```

Comment:
```suggestion
A TCB is created following good development practices. Stabilizing a procedure and including it in a library that is shared in multiple places makes that library part of a TCB. In doing so, ad-hoc trust in that procedure is established because it's been used extensively, tested to exercise edge cases, etc.
```